jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -70,9 +93,45 @@ def process(self, input):
                 'resample_data': self.resample_data,
                 'good_bits': self.good_bits
                 }
+            # Add logic here to select which version of OutlierDetection 
+            # needs to be used depending on the input data
+            exptype = input_models[0].meta.exposure.type 
+            
+            if exptype in IMAGE_MODES:
+                # default mode: imaging with resampling
+                detection_step = outlier_registry['imaging']
+                pars['resample_suffix'] = 'i2d'
+            elif exptype in TSO_SPEC_MODES: 
+                # algorithm selected for TSO data (no resampling)
+                pars['resample_data'] = False # force resampling off...
+                detection_step = outlier_registry['imaging']
+                pars['resample_suffix'] = 'i2d'
+            elif exptype in TSO_IMAGE_MODES+CORON_IMAGE_MODES:
+                # algorithm selected for TSO data (no resampling)
+                pars['resample_data'] = False # force resampling off...
+                detection_step = outlier_registry['imaging']
+                pars['resample_suffix'] = 's2d'                
+            elif exptype in TSO_IMAGE_MODES and self.scale_detection:
+                # selected scaled algorithm for TSO data 
```

Comment:
TSO images will never have any dithering, hence there's no need or desire for resampling. Also, will this `elif` clause ever get reached? If the exposures are TSO_IMAGE_MODES, won't the previous `elif` clause have already gotten triggered and hence there's never a chance of reaching this one, regardless of the value of `self.scale_detection`?

Reply:
I will have to go over this logic again given your comments and see whether this needs to be cleaned up or revised.